components/sidebar/sidebar_channel_list/sidebar_channel_list.tsx

```diff
@@ -370,8 +377,14 @@ export default class SidebarChannelList extends React.PureComponent<Props, State
         if (this.props.categories.some((category) => category.id === before.draggableId)) {
             draggingState.type = DraggingStateTypes.CATEGORY;
         } else {
-            const draggingChannel = this.props.displayedChannels.find((channel) => channel.id === before.draggableId);
-            draggingState.type = (draggingChannel?.type === General.DM_CHANNEL || draggingChannel?.type === General.GM_CHANNEL) ? DraggingStateTypes.DM : DraggingStateTypes.CHANNEL;
+            const draggingChannels = this.props.displayedChannels.filter((channel) => this.props.selectedChannelIds.indexOf(channel.id) !== -1 || channel.id === before.draggableId);
```

Comment:
0/5, but is there maybe a way to do this without looping through the array so many times?

Reply:
We could do a `reduce` to compile an object such that the call to `indexOf` can be replaced with an object lookup, but I don't think it's a huge performance gain.